github/github.go

```diff
@@ -313,7 +313,9 @@ func NewEnterpriseClient(baseURL, uploadURL string, httpClient *http.Client) (*C
 	if !strings.HasSuffix(baseEndpoint.Path, "/") {
 		baseEndpoint.Path += "/"
 	}
-	if !strings.HasSuffix(baseEndpoint.Path, "/api/v3/") {
+	if !strings.HasSuffix(baseEndpoint.Path, "/api/v3/") &&
+		!strings.HasPrefix(baseEndpoint.Host, "api.") &&
```

Comment:
It's never going to have this prefix because of the leading `https://`.

I'm not sure I'm happy yet with this PR because I'm not sure how many users of this repo we are helping or how many we are making more work for (to workaround the code we are adding).

Reply:
Thank you for your response. 

`baseEndpoint.Host` won't contain `http` or `https`. It will just contain this highlighted part: 
 https://**api.domain.com**/query=abc
Example -> [https://play.golang.org/p/DaKtXOVQmKR](https://play.golang.org/p/DaKtXOVQmKR
)

The existing enterprise users won't have to do anything because they will be unaffected if they were not using the `api` subdomain in first place. We would be appending `api/v3/` to their URL, just as we do it today.

However, we would be helping new enterprise users (they won't have to do changes as had to be done by the issue reporter) and also helping existing enterprise users (who were not happy with this breaking change). 

For the enterprise users wich have URLs like:

_https://api.domain.com_ or _https://something.api.domain.com_ ,
we won't append suffix `api/v3/` 

For all other combinations of subdomains, we will continue to append suffix `/api/v3/` .